### ISSUE TITLE
PI 4 HQ Camera

### ISSUE THREAD

**Issue by paveltyk** [electricshaman/picam#52]
Hi,

Thank you for such a great package.

I wonder if that's possible to add support for PI4 and their new HQ camera?

Thank you,
   Pavel.

**Comment by edwardzhou** [comment]
@GregMefford @fhunleth  Hi Guys, is there any plan on this? or any alternative library?

**Comment by fhunleth** [comment]
At the moment, I don't have time to implement an alternative library, but we do need one that uses `libcamera`. `picam` uses the MMAL camera interface and that's going to be harder for us to support in the core Nerves systems as Raspberry Pi OS moves away from it. 

If you haven't already seen https://www.raspberrypi.com/documentation/accessories/camera.html, it has more information on the switch from the old camera software APIs to the new ones. 

While I don't have time to write a camera library now, I'll definitely assist on getting `libcamera` working. From my experiments, `libcamera` is a little tricky to use due to rapid upstream development changing APIs, but it's totally workable. If someone can get an Elixir library that works on Raspberry Pi OS and can point me to it, I'll do the Nerves/RPi4 part to make it work if that's needed.

**Comment by GregMefford** [comment]
Yeah same here. I don’t have any time to invest in building this anytime soon, but I’d love to see it happen and I’d love an excuse to buy one of the new HQ cameras. 🤓
